### DISCUSSION TITLE
Permit null-conditional method group

### DISCUSSION THREAD

**Discussion by RikkiGibson** [dotnet/csharplang#6609] [Language Ideas]
Follow-up to https://github.com/dotnet/roslyn/issues/53860

The following is currently disallowed:
```cs
using System;
using System.Collections.Generic;

class C
{
    // ❌ CS0023 Operator '?' cannot be applied to operand of type 'method group'
    //                             ↓
    Action<int> M(List<int> p) => p?.Add;
}
```

It would be nice if this just worked.

The language in csharpstandard has changed since I wrote the comment https://github.com/dotnet/roslyn/issues/53860#issuecomment-854165001. It would be handy if someone could dig in and figure out whether the standard disallows `Action? a = obj?.Method` and what language would be needed to allow it with the intuitive meaning (`Action a = obj == null ? null : obj.Method`).

**Comment by jnm2** [comment]
The reworked spec is here: https://github.com/dotnet/csharpstandard/blob/standard-v6/standard/expressions.md#1177-null-conditional-member-access

I'm spotting nothing like the line that was quoted from the original spec:

> If E0 is classified as nothing, then E is classified as nothing. Otherwise E is classified as a value.

In the process of raising an issue at https://github.com/dotnet/csharpstandard/issues/654 in case this was an oversight, we realized that this whole thing might turn on the fact that `? null : obj.MethodGroup` started being valid only recently. The rewritten spec defines null-conditional expressions via transformations to conditional expressions and says nothing in particular about method groups or expression classifications.

If this is accurate, then this language request is already implemented in the spec via the change that made that conditional expression valid, and the next step is a bugfix PR to the compiler.
